Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,143 @@ static void Main(string[] args)
+                            case 'P':
+                                FindIndex(protein, index,"max");
+                                break;
+                            case 'p':
+                                FindIndex(protein, index,"min");
+                                break;
+                            case 'F':
+                                FindIndex(fat, index,"max");
+                                break;
+                            case 'f':
+                                FindIndex(fat, index,"min");
+                                break;
+                            case 'T':
+                                FindIndex(calorie, index,"max");
+                                break;
+                            case 't':
+                                FindIndex(calorie, index,"min");
+                                break;
+                        }
+                    }
+                }
+                dietList.Add(index[0]);
+            }
+            int[] diet = new int[dietPlans.Length];
+            dietList.CopyTo(diet);
+            dietList.Clear();
+            return diet;
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        private static void FindIndex(int[] arr, List<int> index,string typeOfIndex)
+        {
+            var list = new List<int>();
+            int value;
+            if(typeOfIndex.Equals("min"))
+            {
+                value = int.MaxValue;
+            }
+            else
+            {
+                value = int.MinValue;
+            }
+            if(index.Count==0)
+            {
+                list.Clear();
+                for(int i=0;i<arr.Length;i++)
+                {
+                    if(value == arr[i])
+                    {
+                        list.Add(i);
+                    }
+                    if(CheckValue(value, arr[i], typeOfIndex))
+                    {
+                        list.Clear();
+                        value = arr[i];
+                        list.Add(i);
+                    }
+                }
+            }
+            else
+            {
+                list.Clear();
+                foreach(var i in index)
+                {
+                    if (value == arr[i])
+                    {
+                        list.Add(i);
+                    }
+                    if (CheckValue(value,arr[i],typeOfIndex))
+                    {
+                        list.Clear();
+                        value = arr[i];
+                        list.Add(i);
+                    }
+                }
+            }
+            index.Clear();
+            for (int i = 0; i < list.Count; i++)
+                index.Add(list[i]);
+        }
+        
+        private static bool CheckValue(int value1,int value2,string operation)

```

Comment:
What type of value is checking this method? Method name should be meaningful. Please correct it.  what is value1, value2?? correct the variable names?